data/cookbook/regex-validate-email/00-str_re.ml

```diff
@@ -0,0 +1,70 @@
+---
+packages:
+  - name: "re"
+    tested_version: "1.11.0"
+    used_libraries:
+      - re
+discussion: |
+  - **Str Example:** Str provides basic regex support but is not recommended given the advantages of the re package. Note: there is no support for regex patterns like \d and \w.
+  - **Advantages of `re`:** Re provides better support for regex syntax encountered in other languages such as python.
+---
+
+(* Sample data for testing *)
+let emails =
+  [| "oscar.0camel@ocaml.org"
+   ; "notAnEmail@jkorg"
+   ; "john.doe@yahoo.com"
+   ; "tommy_trojan@usc.edu"
+   ; "emailScammer@evil.lol"
+  |]
+;;
+
+(* Function to test patterns *)
+let test_pattern validation_func emails =
+  print_endline "Email Verification Results:";
+  Array.map (fun x -> validation_func x) emails
+;;
+
+(* Using Str package *)
+let validation_pattern =
+  Str.regexp {regexp|[A-Za-z0-9]+.?[A-Za-z0-9]+@[a-zA-Z]+\.[a-za-z-a-z]|regexp}
+;;
+
+(* `validate_email_str` accepts a string `email`
+   and checks it against the validation pattern *)
+let validate_email_str email =
+  if Str.string_match validation_pattern email 0
+  then Printf.printf "%s has a valid email format\n" email
+  else Printf.printf "%s has an invalid email format\n" email
+;;
+
+(* Print test results to console *)
+test_pattern validate_email_str emails
+
+(* Using the Re package *)
```

Comment:
Could probably have a better comment here along the lines that we're defining the regular expression pattern

Reply:
Thank you @sabine! I made some changes and added better comments.